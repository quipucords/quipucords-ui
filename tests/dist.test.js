const { execSync } = require('child_process');

describe('Distribution', () => {
  const trimSortDistList = str =>
    str
      .replace(/\s+|\n+|\r+/g, '')
      .replace(/\.\/dist/gi, '~./dist')
      .replace(/\.([a-z0-9]+)\./gi, '*')
      .split('~')
      .sort();

  it('should match a specific file output', () => {
    const output = execSync('find ./dist -type f -print0 | xargs -0').toString();

    expect(trimSortDistList(output)).toMatchSnapshot('dist-output');
  });

  it('should not contain references to localhost', () => {
    const output = execSync(`grep -roi "localhost:" ./dist/client/static | wc -l`).toString();

    expect(Number.parseInt(output.trim(), 10)).toBe(0);
  });
});
