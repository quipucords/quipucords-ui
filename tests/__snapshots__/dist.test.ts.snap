// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build distribution should match a specific file output: dist-output 1`] = `
[
  "",
  "./build/113*js",
  "./build/362*css",
  "./build/362*js",
  "./build/820*js",
  "./build/99*js",
  "./build/app*css",
  "./build/app*js",
  "./build/favicon.ico",
  "./build/fonts/078865af129336cde269.woff2",
  "./build/fonts/18ca482120faaa701d8b.woff2",
  "./build/fonts/57234ce675330fa60781.woff2",
  "./build/fonts/66ac08a9a7d29ae51581.woff2",
  "./build/fonts/6e3027616d8de472723c.woff2",
  "./build/fonts/813182e673c38ef5ee4b.woff2",
  "./build/fonts/841cdd0ebb3d9038a7e7.woff2",
  "./build/fonts/9c02e9a1e10fbc86466d.woff2",
  "./build/fonts/aa07940756a803076b34.woff2",
  "./build/fonts/b01a8365f7214919a9c3.woff2",
  "./build/fonts/b36d694117bf12672d49.woff2",
  "./build/fonts/c06f4d6490c387fd1d92.woff2",
  "./build/fonts/c33624692c8b41e36780.woff2",
  "./build/fonts/d8d66dd2281610675aca.woff2",
  "./build/fonts/e0f30d27bec7c8dbb6fc.woff2",
  "./build/fonts/e4faf134e53e0f37296b.woff2",
  "./build/fonts/ed292ec87cce0d76fd3c.woff2",
  "./build/images/133911d429584887f5c7.svg",
  "./build/images/4ca682d556b7bf59f5c6.svg",
  "./build/images/c0ce9e7676e8e8f83167.png",
  "./build/index.html",
  "./build/locales/en.json",
  "./build/locales/locales.json",
  "./build/manifest.json",
  "./build/runtime*js",
  "./build/vendor*css",
  "./build/vendor*js",
  "./build/vendor*txt",
]
`;
